Src/Support/Google.Apis.Auth/SignedTokenVerification.cs

```diff
@@ -130,15 +141,66 @@ static RSA FromKeyToRsa(JToken key)
             }
         }
 
-        private static Task VerifyES256TokenAsync<TJswHeader, TJswPayload>(
+#if NETSTANDARD2_0 || NET45
+        private async static Task VerifyES256TokenAsync<TJswHeader, TJswPayload>(
             SignedToken<TJswHeader, TJswPayload> signedToken, SignedTokenVerificationOptions options, CancellationToken cancellationToken)
             where TJswHeader : Header
             where TJswPayload : Payload
         {
-            throw new NotImplementedException();
+            byte[] hash;
+            using (var hashAlg = SHA256.Create())
+            {
+                hash = hashAlg.ComputeHash(Encoding.ASCII.GetBytes($"{signedToken.EncodedHeader}.{signedToken.EncodedPayload}"));
+            }
+
+            var certificates = await(options.CertificateCache ?? s_certificateCache).GetCertificatesAsync(
+                options.CertificatesUrl ?? GoogleAuthConsts.IapKeySetUrl,
+                FromKeyToECDsa,
+                options.ForceCertificateRefresh,
+                cancellationToken).ConfigureAwait(false);
+            foreach (ECDsa certificate in certificates)
+            {
+                if (certificate.VerifyHash(hash, signedToken.Signature))
+                {
+                    return;
+                }
+            }
+            throw new InvalidJwtException("JWT invalid, unable to verify signature.");
+
+            static ECDsa FromKeyToECDsa(JToken key)
+            {
+                if ((string)key["kty"] != "EC" && (string)key["crv"] != "P-256")
+                {
+                    throw new ArgumentException(
+                        $"For ES256 verification only certificates with kty='EC' and crv='P-256' are supported. Encountered: kty={(string)key["kty"]} and crv={(string)key["crv"]}.");
+                }
+                byte[] x = TokenEncodingHelpers.Base64UrlDecode((string)key["x"]);
+                byte[] y = TokenEncodingHelpers.Base64UrlDecode((string)key["y"]);
+#if NETSTANDARD2_0
```

Comment:
Could this be extracted to a separate method? I think it would end up being a bit clearer that way.